decoder/vaapidecoder_vp9.cpp

```diff
@@ -90,23 +91,50 @@ void VaapiDecoderVP9::flush(void)
 
 Decode_Status VaapiDecoderVP9::ensureContext(const Vp9FrameHdr* hdr)
 {
+    bool resetContext = false;
+    Decode_Status status = DECODE_SUCCESS;
+    DEBUG("got frame size: %d x %d", hdr->width, hdr->height);
+
     // only reset va context when there is a larger frame
     if (m_configBuffer.width < hdr->width
         || m_configBuffer.height <  hdr->height) {
+        resetContext = true;
         INFO("frame size changed, reconfig codec. orig size %d x %d, new size: %d x %d",
                 m_configBuffer.width, m_configBuffer.height, hdr->width, hdr->height);
-        Decode_Status status = VaapiDecoderBase::terminateVA();
+        
+        if (m_hasContext){
+            status = VaapiDecoderBase::terminateVA();
```

Comment:
yes, this patch only fix "display lost" at beginning of playback. If the bitstream have resolution change in the middle of playback. It still lost display. So i proposed other patch to sameer. I planed to test this patch after we get my image ready. But I stucked at hang issue. So it's great if sammer can help me test this patch and push to this thread.It can help us move faster.